proposals/1533-host-specific-tasklist.md

```diff
@@ -0,0 +1,241 @@
+
+### Service Deployment
+
+In this case, a session represents the deployment of a service to one worker and the number of concurrent deployments is critical. User can have control over that number by limiting the number of concurrently running sessions on an activity worker.
+
+## Proposal
+
+The scope of the proposal contains only Cadence client. Cadence server doesn’t need to be changed at all. The basic idea here is that we should write special workflow code for users to perform activity scheduling and worker monitoring, and expose simple APIs to users. Now that users don’t need to care about those implementation details, they can focus on their own business logic.
+
+The proposal includes both API changes and implementations. It aims at solving three problems:
+
+1. How to ensure multiple activities are executed on one activity worker?
+
+2. How to limit the number of concurrent sessions on an activity worker?
+
+3. How to carry over session information between different workflow runs?
+
+The following sections will first go through the APIs exposed to users, provide some explanation on how to use them and some sample code. It should be very easy for any developer with previous cadence experience to understand those APIs and start using them. After that I will explain the model behind those APIs and how they are implemented.
+
+### API Changes
+
+#### New Go Client API
+
+There will be four new APIs available when writing a workflow:
+
+```go
+sessionCtx, err := workflow.**CreateSession**(ctx)
+```
+
+User call this API to create a session and use the returned sessionCtx to execute activities. All activities executed within returned sessionCtx are considered to be part of the session and will be executed on the same worker. It’s possible that this call will fail due to various reasons: for example, no worker is available at this time.
+
+The session will be created on one of the workers that poll from the tasklist specified in the context passed into the CreateSession API. As a result, users needs to make sure they are using the correct task list.
+
+When executing an activity within a session, user may get two types of errors. The first one is those returned from user activities. Session will not be marked as failed in this case, so user can return whatever error they want and apply their business logic as necessary. If they want to end a session due to the error returned from the activity, use the CompleteSession() API below. The second type of error is a special error which says the session has failed due to worker failure and the session is marked as failed in the background. In this case, no activities can be executed using this context. It’s the user’s choice on how to handle the failure; for example, they can create a new session to retry or end the workflow with an error.
```

Comment:
a session, user may -> a session, a user may
The first one is those returned -> The first one is returned
Session will not -> The session will not
so user can -> so the user can
If they want -> If a user wants 

It's the user's choice on how to handle the failure; for example, they can  -> The user can choose how to handle the failure. They can